File: uniclub-hub.client/src/lib/signalrEvents.ts
export const SIGNALR_EVENTS = {
  TASK_STATUS_UPDATED:  'TaskStatusUpdated',
  TASK_CREATED:         'TaskCreated',
  TASK_DELETED:         'TaskDeleted',
  EVENT_TASKS_CLEANED:  'EventTasksCleaned',  // fired after cascade soft-delete
  NOTIFICATION_RECEIVED: 'NotificationReceived',  // per-user in-app notification
  COMMENT_ADDED:         'CommentAdded',         // new comment on a task
  ATTACHMENT_UPLOADED:   'AttachmentUploaded',   // new attachment on a task
  SPRINT_STATUS_CHANGED: 'SprintStatusChanged',  // sprint status transition
  ASSIGNMENT_RECEIVED:   'AssignmentReceived',   // club received a new assignment
  ASSIGNMENT_CANCELLED:  'AssignmentCancelled',  // an assignment was cancelled by the university admin
} as const

export const HUB_METHODS = {
  JOIN_CLUB:  'JoinClub',
  LEAVE_CLUB: 'LeaveClub',
} as const

File: UniClub-Hub.Shared/Constants/SignalREvents.cs
namespace UniClub_Hub.Shared.Constants;

public static class SignalREvents
{
    public const string TaskStatusUpdated  = "TaskStatusUpdated";
    public const string TaskCreated        = "TaskCreated";
    public const string TaskDeleted        = "TaskDeleted";
    public const string EventTasksCleaned  = "EventTasksCleaned";  // broadcast when cascade-delete fires
    public const string NotificationReceived = "NotificationReceived";  // per-user in-app notification
    public const string CommentAdded         = "CommentAdded";          // new comment on a task
    public const string AttachmentUploaded   = "AttachmentUploaded";    // new attachment on a task
    public const string SprintStatusChanged  = "SprintStatusChanged";   // sprint status transition
    public const string AssignmentReceived   = "AssignmentReceived";    // club received a new work assignment
}

public static class SignalRGroups
{
    public static string Club(int clubId) => $"club_{clubId}";
}
